java/jdbc/src/main/java/io/vitess/jdbc/VitessJDBCUrl.java

```diff
@@ -158,11 +158,22 @@ public VitessJDBCUrl(String url, Properties info) throws SQLException {
             throw new SQLException(Constants.SQLExceptionMessages.MALFORMED_URL);
         }
 
-        this.keyspace = StringUtils.isNullOrEmptyWithoutWS(m.group(8)) ? null : m.group(8);
+        this.keyspace = StringUtils.isNullOrEmptyWithoutWS(m.group(8)) ? info.getProperty(Constants.Property.KEYSPACE) : m.group(8);
```

Comment:
I think we should remove these altogether and make all property access go through ConnectionProperties. Only reason I didn't do that when I originally created ConnectionProperties was I was trying to avoid too much changes. But now we are bumping the version and doing a lot of changes, so might be worthwhile.

I already recommended in another comment to create a ConnectionProperties.getKeyspace(). I'd also recommend removing the getters from this class.  This would be finalizing the plan to make VitessJDBCUrl just the URL parser and all property-related accessor code happen in ConnectionProperties.